### PR TITLE
Fixes beam runtimes

### LINKED ISSUE #12336: Crippling emitter/blob issues
(WEB REPORT BY: sprok REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> b798ff8691b1c79c88d6b633cee3f113736d3967
> 
> General description of the issue
> Emitter beams, after destroying a tile of blob, stop in place and don't move forward to the next. If a blob expands into the now empty tile, it does not get damaged by the beam next to it.
> Normal blobs that are being damaged by an emitter, when turned into strong blobs, stop taking damage.
> 
> What you expected to happen
> Emitters to bore holes through blobs
> 
> What actually happened
> They don't
> 
> Steps to reproduce if possible
> Spawn a blob core, point emitters at it

### PR DESCRIPTION
**Pull request by BarneyGumball**
Sanity checks and removing the unnecessary damage application on beam disconnect for blobs.  The blob piece would try to delete itself twice resulting in runtimes if it was destroyed with a beam attached.

Fixes #12336

:cl:
- bugfix: Fixed emitters bugging out on blobs.

### CODE REVIEW

File: code/game/gamemodes/blob/theblob.dm [hunk 1]
```diff
@@ -148,9 +148,7 @@ var/list/blob_looks
 
 /obj/effect/blob/beam_disconnect(var/obj/effect/beam/B)
 	..()
-	apply_beam_damage(B)
```
Comment (clusterfack) [inline code/game/gamemodes/blob/theblob.dm]: ???

File: code/__HELPERS/_macros.dm [hunk 2]
```diff
@@ -133,6 +133,8 @@
 
 #define isatom(A) (istype(A, /atom))
 
+#define isatommovable(A) (istype(A, /atom/movable))
```
Comment (clusterfack) [inline code/__HELPERS/_macros.dm]: Now these are just getting silly

Comment (gbasood) [reply code/__HELPERS/_macros.dm:136]: `define #willitblend`

**Comment by FudgePucker** [comment]
How would a gyrotron do against blob compared to emitters? It is like an old emitter in that it shoots out projectiles instead of a beam.

**Comment by PJB3005** [comment]
@fudgepucker uuuuuuuh.... That's a really good question considering the gyrotrons shoot multiple times faster than emitters.

File: code/game/gamemodes/blob/theblob.dm (hunk 1 above)
Comment (gbasood) [reply code/game/gamemodes/blob/theblob.dm:151]: `removing the unnecessary damage application on beam disconnect for blobs. The blob piece would try to delete itself twice resulting in runtimes if it was destroyed with a beam attached.`

Comment (clusterfack) [inline code/game/gamemodes/blob/theblob.dm]: Yeah the problem is its supposed to damage normal things when they move out of the way so they don't out time it though? You could as easily check gcdestroyed/disposed cant you

Comment (BarneyGumball) [reply code/game/gamemodes/blob/theblob.dm:151]: For some reason, I thought this had the bonus of preventing an exploit to get extra damage on blobs.

Coding while tired is probably a bad idea. :thinking:

**Comment by BarneyGumball** [comment]
Fixed it properly and removed a bunch of copy-paste.

**Pull request merged**